File: usecase-07/src/main/ruby/TextUtils.rb
require "base64"

def decode64(encodedText)
  return Base64.decode64(encodedText)
end

def extractWords(text)
  plainText = decode64(text)

  # puts plainText
  # puts File.basename(Dir.getwd)
  #
  pwd = File.expand_path(File.dirname(__FILE__))
  Dir.chdir pwd
  puts pwd

  Polyglot.eval_file("js", "../javascript/TextUtils.js")

  sortWords = Polyglot.import_method("sortWords")

  return sortWords(plainText)
end

Polyglot.export_method("extractWords")

File: usecase-07/src/main/javascript/TextUtils.js
function sortWords(text) {
    const result = [];
    const words = text.split(' ');

    words.sort();

    for (const element of words) {
        result.push(element.replace(/[^a-zA-Z]/g, ""));
    }

    return result;
}

Polyglot.export('sortWords', sortWords);
